### ISSUE TITLE
Improve homepage accessibility

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary#11353]
### Feature Request

* The `<div id="test-body-mobile">` should be a `<main>` element to allow it to be accessible as a landmark
* The images in the onboarding carousel should be tagged with `alt=""` to note that they are purely presentational and should not be read out by a screen reader
* The `<h3 id="home-about-mission-tldr">` should not be an `h3`, it should just be a `<p>`. It causes confusing heading navigation with a screen reader
* "Latest blog posts" should be an `<h3>`
* The images in the "Browse by subject carousel" should have `alt=""`; they are presentational and should not be read out by screenreaders
* The form that contains the search box should have `role="search"`

### Related files

Refer to [this map of common Endpoints](https://github.com/internetarchive/openlibrary/wiki/Endpoints):
*

### Requirements Checklist
Checklist of requirements that need to be satisfied in order for this issue to be closed:
* [ ]

### Stakeholders

*

<hr>

### Instructions for Contributors

* **Before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) or pushing up changes to a PR, please first [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date, as the pre-commit bot may add commits to your PRs upstream.
